### PR TITLE
feat: [KRP-1055] create form for postbox item

### PR DESCRIPTION
**Pull request by haipham-kontist**
- Create a form to send postbox item to backend.

Questions:
- I can't call `createWebhook` function directly via postman due to auth issue. Is it intended? There's no document about that. What I did was to disable the auth check.
- Can't make request to backend due to `FetchError: request to http://localhost:3000/api/webhook/solaris/postbox-item-created failed, reason: connect ECONNREFUSED 127.0.0.1:3000`. I think because it's not in the same network. How do I apply this one to `mock-solaris` project in `backend`? (The route is available in backend)

### CODE REVIEW

**Comment by p-janik** [comment]
@haipham-kontist this looks good. You can now set dev/staging secrets in our backend repo.
You can work locally with this project by adjusting `docker-compose-dev.yml`, details how to do it you can find in this file.

File: src/templates/person.html [hunk 1]
```diff
@@ -152,6 +152,59 @@ <h2>{{ person.email }}</h2>
...
+            <input class="form-control" type="text" name="name" placeholder="Document name" autocomplete="off" required />
+          </div>
+          <div class="form-group">
+            <label for="description">Document description</label>
+            <input class="form-control" type="text" name="description" placeholder="Document description" autocomplete="off" required />
+          </div>
+          <div class="form-group">
+            <label for="documentType">Document type</label>
+            <select name="documentType" class="form-control" required>
+              <option {% if person.customer_vetting_status === "ACCOUNT_STATEMENT" %}selected {% endif %} value="ACCOUNT_STATEMENT">ACCOUNT_STATEMENT</option>
```
Comment (p-janik) [inline src/templates/person.html]: nit: you could pass enum to handler and then iterate on it here to create option elements

Comment (p-janik) [reply src/templates/person.html]: is it for sure `customer_vetting_status`?

Comment (haipham-kontist) [reply src/templates/person.html]: removed :+1:

File: src/app.ts [hunk 2]
```diff
@@ -530,6 +531,17 @@ router.post(
...
 );
 
+// POSTBOX ITEM
+router.get(
+  "/persons/:person_id/postbox/items",
+  safeRequestHandler(postboxItemAPI.listPostboxItems)
+);
+
+router.get(
+  "/postbox/items/:postbox_item_id/document",
```
Comment (p-janik) [inline src/app.ts:546]: we can also add `/postbox/items/{item_id}
`

File: src/routes/postbox.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+};
+
+export const listPostboxItems = async (req, res) => {
+  const personId = req.params.person_id;
+  assert(personId, "person ID not found");
+
+  log.info(`listPostboxItems() get list of postbox items for ${personId}`);
+
+  const person = await getPerson(personId);
+  res.status(200).send(person.postboxItems);
```
Comment (p-janik) [inline src/routes/postbox.ts]: shall we fallback to empty array for old users?

File: src/routes/postbox.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+};
+
+export const downloadPostboxItem = async (req, res) => {
+  log.info(
+    `downloadPostboxItem() download postbox items ${req.params.postbox_item_id}`
+  );
+
+  res.download(path.join(__dirname, "../assets/sample.pdf"));
```
Comment (p-janik) [inline src/routes/postbox.ts:144]: let's add 404 in case when `postbox_item_id` is missing

**Review by p-janik — APPROVED** [review]

File: src/app.ts (hunk 2 above)
Comment (haipham-kontist) [reply src/app.ts:546]: that's endpoint is for [getting info of the item](https://docs.solarisgroup.com/api-reference/onboarding/compliance/#operation/solaris_postbox_items_find), not [downloading](https://docs.solarisgroup.com/api-reference/onboarding/compliance/#operation/solaris_postbox_item_document_find) it.

File: src/routes/postbox.ts (hunk 4 above)
Comment (haipham-kontist) [reply src/routes/postbox.ts:144]: done :+1: https://github.com/kontist/mock-solaris/pull/79/commits/0eb2a4c9f59b959b88741c19d052e7a3dd5c56cf

File: src/routes/postbox.ts (hunk 3 above)
Comment (haipham-kontist) [reply src/routes/postbox.ts]: done https://github.com/kontist/mock-solaris/pull/79/commits/0eb2a4c9f59b959b88741c19d052e7a3dd5c56cf

File: src/app.ts (hunk 2 above)
Comment (haipham-kontist) [reply src/app.ts:546]: ah I see what you mean :+1:

Comment (haipham-kontist) [reply src/app.ts:546]: endpoint added https://github.com/kontist/mock-solaris/pull/79/commits/d0ef44a1dd3c6104a5637ed5d7a06dc16e50536f

**Review by p-janik — APPROVED** [review]
thank you 👍

**Pull request merged**